libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -688,26 +687,25 @@ class MapboxNavigation(
 
     private fun configureRouter() {
         with(navigationOptions) {
-            ThreadController.getMainScopeAndRootJob().scope.launch {
```

Comment:
it's good to know that this is causing the sync issue. but i don't think the solution is to access the file system on the main thread - that's trading issues